src/components/PopupEditAvatar.js

```diff
@@ -0,0 +1,38 @@
+import Popup from "./Popup.js";
+
+class PopupEditAvatar extends Popup {
+  constructor(popupSelector, handleFormSubmit) {
+    super(popupSelector);
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._handleFormSubmit = handleFormSubmit;
+  }
+
+  _getInputValues() {
+    const inputValues = {};
+    const popupFormInputs =
+      this._popupForm.querySelectorAll(".modal__form-input");
+    popupFormInputs.forEach((input) => {
+      inputValues[input.name] = input.value;
+    });
+
+    return inputValues;
+  }
+
+  _handleSubmitButton = () => {
+    this._handleFormSubmit(this._getInputValues());
+    this.close();
```

Comment:
Сlose modal in the case of successful api response in `then` block in `index.js`, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.